### PR TITLE
add new fuzz target for ndpi project

### PR DESCRIPTION
**Pull request by koltiradw**

### CODE REVIEW

File: projects/ndpi/Dockerfile [hunk 1]
```diff
@@ -29,10 +29,10 @@ RUN mkdir corpus && mv nDPI/tests/pcap/* corpus
...
-RUN git checkout 4231f4805965544994fa8d5b0d0ec56573c5f5d2
+RUN git checkout 63f349319f00abd29af03ebefc969caf17fa65e9
 
-COPY build.sh load_sydr_process_packet.c load_sydr_ndpi_reader.c ./
+COPY build.sh load_sydr_process_packet.c load_sydr_ndpi_reader.c load_sydr_quic_get_crypto_data.c ./
 
 RUN ./build.sh
 
-WORKDIR /
+WORKDIR /
```
[the remark this answers is not in the document]
Comment (SweetVishnya) [reply projects/ndpi/Dockerfile:38]: and other similar places in diff

File: projects/ndpi/load_sydr_quic_get_crypto_data.c [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+{
+        const u_int8_t *crypto_data;
+        uint64_t crypto_data_len;
+        u_int32_t first_int, version = 0;
+        if(ndpi_info_mod == NULL) {
+                ndpi_info_mod = ndpi_init_detection_module(ndpi_enable_ja3_plus);
+                NDPI_PROTOCOL_BITMASK all;
+                NDPI_BITMASK_SET_ALL(all);
+                ndpi_set_protocol_detection_bitmask2(ndpi_info_mod, &all);
+#if 0
```
Comment (anfedotoff) [inline projects/ndpi/load_sydr_quic_get_crypto_data.c:34]: Maybe remove debug?

**Review by anfedotoff — CHANGES_REQUESTED** [review]

File: projects/ndpi/load_sydr_quic_get_crypto_data.c [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+extern int is_version_with_tls(uint32_t version);
+
+
+int main(int argc, char** argv)
+{
+        const u_int8_t *crypto_data;
+        uint64_t crypto_data_len;
+        u_int32_t first_int, version = 0;
+        if(ndpi_info_mod == NULL) {
+                ndpi_info_mod = ndpi_init_detection_module(ndpi_enable_ja3_plus);
```
[the remark this answers is not in the document]
Comment (SweetVishnya) [reply projects/ndpi/load_sydr_quic_get_crypto_data.c:30]: We can also update commit in git checkout

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: projects/ndpi/load_sydr_quic_get_crypto_data.c [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        fseek(fd, 0, SEEK_END);
+        int Size = ftell(fd);
+        fseek(fd, 0, SEEK_SET);
+
+        unsigned char* Data = (unsigned char*) malloc(sizeof(unsigned char) * Size);
+        fread(Data, 1, Size, fd);
+        fclose(fd);
+
+        if(Size < 4)
+                return 0;
```
[the remark this answers is not in the document]
Comment (koltiradw) [reply projects/ndpi/load_sydr_quic_get_crypto_data.c:57]: oh my fault

[the remark this answers is not in the document]
Comment (koltiradw) [reply projects/ndpi/load_sydr_quic_get_crypto_data.c]: resolved, thanks!

**Review by anfedotoff — APPROVED** [review]

**Comment by anfedotoff** [comment]
I have problems with building docker image:

     configure: error: Missing libpcap(-dev) library required to compile the example application.

Could you check the build?

**Comment by koltiradw** [comment]
> I have problems with building docker image:
> 
> ```
>  configure: error: Missing libpcap(-dev) library required to compile the example application.
> ```
> 
> Could you check the build?



> I have problems with building docker image:
> 
> ```
>  configure: error: Missing libpcap(-dev) library required to compile the example application.
> ```
> 
> Could you check the build?

**Review by anfedotoff — APPROVED** [review]

File: projects/ndpi/build.sh [hunk 5]
```diff
@@ -62,13 +66,17 @@ make install
 
 cd /nDPI
 
-./autogen.sh
+LDFLAGS="-lpcap"./autogen.sh --with-only-libndpi
```
Comment (apach301) [inline projects/ndpi/build.sh:69]: `LDFLAGS="-lpcap" ./autogen.sh --with-only-libndpi`

**Review by apach301 — CHANGES_REQUESTED** [review]

File: projects/ndpi/build.sh (hunk 5 above)
Comment (koltiradw) [reply projects/ndpi/build.sh]: resolved

File: projects/ndpi/build.sh [hunk 6]
```diff
@@ -62,13 +66,17 @@ make install
 
 cd /nDPI
 
-./autogen.sh
+LDFLAGS="-lpcap" ./autogen.sh --with-only-libndpi
```
Comment (SweetVishnya) [inline projects/ndpi/build.sh:69]: We'd better remove build directory, autogen may contain some cache. rm -rf build_dir may be better than make clean

Comment (koltiradw) [reply projects/ndpi/build.sh:69]: Every run of autogen.sh call autoreconf - re-creating configure and makefiles.

**Pull request merged**